src/jit/flowgraph.cpp

```diff
@@ -4222,6 +4222,16 @@ void Compiler::fgFindJumpTargets(const BYTE* codeAddr,
         compInlineResult->NoteBool(InlineObservation::CALLEE_IS_FORCE_INLINE, isForceInline);
         compInlineResult->NoteInt(InlineObservation::CALLEE_IL_CODE_SIZE, codeSize);
 
+#ifdef DEBUG
+
+        // If inlining, this method should still be a candidate.
+        if (compIsForInlining())
+        {
+            assert(compInlineResult->IsCandidate());
+        }
```

Comment:
you could do this on one line, without #ifdef, with:
  assert(!compIsForInlining() || compInlineResult->IsCandidate());
just to reduce the number of #ifdefs in the code.